tests/sentry/api/endpoints/test_organization_api_key_details.py

```diff
@@ -31,13 +33,38 @@ class OrganizationApiKeyDetailsPut(OrganizationApiKeyDetailsBase):
     method = "put"
 
     def test_update_api_key_details(self):
-        data = {"label": "New Label", "allowed_origins": "sentry.io"}
+        data = {
+            "label": "New Label",
+            "allowed_origins": "sentry.io",
+            "scope_list": ["a", "b", "c", "d"],
+        }
         self.get_success_response(self.organization.slug, self.api_key.id, **data)
 
         api_key = ApiKey.objects.get(id=self.api_key.id, organization_id=self.organization.id)
 
         assert api_key.label == "New Label"
         assert api_key.allowed_origins == "sentry.io"
+        assert api_key.get_scopes() == ["a", "b", "c", "d"]
+
+    def test_update_api_key_details_legacy_data(self):
+        # Some old api keys have this psql special format string
+        self.api_key.scope_list = "{event:read,member:read,org:read,project:read,team:read}"
```

Comment:
Postgres docs mention that values in the set can have quotes on them. Hopefully we don't need to handle those kinds of values as well. If so, we can make further changes.

Reply:
That's true.  Technically, I would need bindings to postgres libs to have a complete parser to do this 100% correctly.  There are libraries for python that do so, but the cost is pulling in another lib for pypi.

In reality I don't think this is an issue because the kinds of historical values that were allowed in the affected columns themselves never need quoting.

Reply:
👍 to keeping it simple for now.